components/mosart/src/cpl/rof_comp_esmf.F90

```diff
@@ -733,10 +754,21 @@ subroutine rof_import_esmf( x2r_array, rc)
           shum               = fptr(index_x2r_Sa_shum,n2)
           THeat%forc_vp(n)   = shum * THeat%forc_pbot(n)  / (0.622_r8 + 0.378_r8 * shum)
           THeat%coszen(n) = fptr(index_x2r_coszen_str,n2)
-       end if                 
+       end if
+
+       rtmCTL%qsur(n,nmud) = 0.0_r8
+       rtmCTL%qsur(n,nsan) = 0.0_r8                 
 
     enddo
 
+    if(sediflag) then
+        do n = begr,endr
+           n2 = n - begr + 1
+           rtmCTL%qsur(n,nmud) =  fptr(index_x2r_Flrl_rofmud,n2) * (rtmCTL%area(n)) ! kg/m2/s --> kg/s for sediment
```

Comment:
Do we need to account for the land fraction of the grid cell? For example, a fraction of the grid cell is land for coastal grid cell. Or does rtmCTL%area already account for the land fraction?

Reply:
You can ignore the _esmf.F90 file.  It is not used in E3SM.

Reply:
In addition to what @rljacob put there, we are doing it similarly to rtmCTL%qsur and rtmCTL%qsub in this subroutine, i.e., not accounting for the land fraction here. @donghuix, you are welcome to investigate it further, but I'd suggest that it is beyond the scope of this PR, and let's go with what we have for now and have it merged into the master branch. In case your investigation does suggest we need to account for the land fraction, you may later issue another PR for it.

Reply:
Yes, I agree. It should be consistent with rtmCTL%qsur. I don't have other comments for this PR. Thanks.